internal/application/callback.go

```diff
@@ -256,10 +256,12 @@ func updateAssociatedProfile(profileName string, dic *di.Container) errors.EdgeX
 			errMsg := fmt.Sprintf("failed to add profile %s", profileName)
 			return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
 		}
+		return nil
 	}
 	err = cache.Profiles().Update(dtos.ToDeviceProfileModel(res.Profile))
 	if err != nil {
 		lc.Warnf("failed to to update profile %s in cache, using the original one", profileName)
+		return err
```

Comment:
please use the EdgeX error type to wrap the error here

```
errMsg := fmt.Sprintf(failed to to update profile %s in cache, using the original one", profileName)
return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
```

Reply:
ok